### PR TITLE
Fix for issue #193

### LINKED ISSUE #193: Default database 0 overrides StackExchange.Redis defaultDatabase in configuration string
**Version Information**
Version of Akka.NET? 1.4.31
Which Akka.NET Modules? Akka.Persistence.Redis

**Describe the bug**
Akka.Persistence.Redis describes using the `configuration-string` using the [StackExchange.Redis configuration string format](https://stackexchange.github.io/StackExchange.Redis/Configuration#basic-configuration-strings). One option of the configuration of StackExchange.Redis is `defaultDatabase`, which selects the database the library should use by default.

This configuration setting is overridden by the Akka.Persistence.Redis setting `database`, which is undocumented (at least in the README) and defaults to 0.

As such, attempting to set the database using the configuration string does not work, always results in using database 0 and is not immediately clear why.

I suggest that this setting `database` be documented. Additionally, if `database` is unspecified, I think it shouldn't override the configuration string-specified `defaultDatabase` (however I can see this might inadvertently break existing deployments if they have used a non-zero `defaultDatabase` but are, in fact, using database 0).

### PR DESCRIPTION
**Pull request by eaba**
Fixes #193

## Changes

made `defaultDatabase` aware

### CODE REVIEW

File: src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            DbUtils.Initialize(fixture);
+
+            return ConfigurationFactory.ParseString($@"
+            akka.loglevel = INFO
+            akka.persistence.journal.plugin = ""akka.persistence.journal.redis""
+            akka.persistence.journal.redis {{
+                class = ""Akka.Persistence.Redis.Journal.RedisJournal, Akka.Persistence.Redis""
+                plugin-dispatcher = ""akka.actor.default-dispatcher""
+                configuration-string = ""{DbUtils.ConnectionString},defaultDatabase=0""
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs:35]: Set this to a non-zero value to ensure that it's picked.

Comment (Aaronontheweb) [reply src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs:35]: And then validate that we're talking to the non-zero database.

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Haven't reviewed all of it right way but spotted an immediate test fragility issue.

Don't use `defaultDatabase=0` for any of these tests - can't tell if the new code is working or the old defaults are if that's the case.

File: src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs]: And is that a function of how we're using the driver?

Comment (Aaronontheweb) [reply src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs]: Or, should we support multiple servers? I thought cluster mode took care of the "scale out" use case already.

Comment (eaba) [reply src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs]: It appears we will have to setup multiple servers explicitly. Right now it is only one server - `0`!

Comment (eaba) [reply src/Akka.Persistence.Redis.Cluster.Tests/RedisJournalDefaultDatabaseSpec.cs]: Multiple databases not supported with redis cluster according to this comment https://github.com/StackExchange/StackExchange.Redis/issues/1188#issuecomment-513149394

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**